include/nbl/builtin/hlsl/prefix_sum_blur.hlsl

```diff
@@ -0,0 +1,262 @@
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+#include "nbl/builtin/hlsl/glsl_compat/core.hlsl"
+#include "nbl/builtin/hlsl/workgroup/basic.hlsl"
+#include "nbl/builtin/hlsl/workgroup/arithmetic.hlsl"
+#include "nbl/builtin/hlsl/workgroup/scratch_size.hlsl"
+#include "nbl/builtin/hlsl/device_capabilities_traits.hlsl"
+#include "nbl/builtin/hlsl/enums.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace box_blur
+{
+
+template<
+    typename DataAccessor,
+    typename SharedAccessor,
+    typename ScanSharedAccessor,
+    typename Sampler,
+    uint16_t WorkgroupSize,
+    class device_capabilities=void> // TODO: define concepts for the Box1D and apply constraints

```

Comment:
okay, so in C++ you'd have the option of a `template<class> class` parameter for the `Sampler` but its of limited usefulness anyway.

Instead what you can require is that `Sampler::prefix_sum_accessor_t` exists, and then you don't need to ask for the type of `SharedAccessor` in the template at all.

Even better yet, you could add a `prefixSumAccessor` member to `Sampler` and then you don't need to pass `scratch` to `Box1D::operator()` at all as an argument (they agree).